File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/operations/get_chargemetrics.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from ..shared import chargemetricsresponse as shared_chargemetricsresponse
from ..shared import commonreasonserrorresponse as shared_commonreasonserrorresponse


@dataclasses.dataclass
class GetChargeMetricsQueryParams:
    from_timestamp: datetime = dataclasses.field(metadata={'query_param': { 'field_name': 'fromTimestamp', 'style': 'form', 'explode': True }})
    to_timestamp: datetime = dataclasses.field(metadata={'query_param': { 'field_name': 'toTimestamp', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetChargeMetricsHeaders:
    accept: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Accept', 'style': 'simple', 'explode': False }})
    zuora_entity_ids: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Entity-Ids', 'style': 'simple', 'explode': False }})
    zuora_track_id: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Track-Id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetChargeMetricsRequest:
    headers: GetChargeMetricsHeaders = dataclasses.field()
    query_params: GetChargeMetricsQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class GetChargeMetricsResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    charge_metrics_response: Optional[shared_chargemetricsresponse.ChargeMetricsResponse] = dataclasses.field(default=None)
    common_reasons_error_response: Optional[shared_commonreasonserrorresponse.CommonReasonsErrorResponse] = dataclasses.field(default=None)
    

File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/operations/getchargemetrics.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetChargeMetricsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fromTimestamp" })
  fromTimestamp: Date;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=toTimestamp" })
  toTimestamp: Date;
}


export class GetChargeMetricsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Accept" })
  accept?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Entity-Ids" })
  zuoraEntityIds?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Track-Id" })
  zuoraTrackId?: string;
}


export class GetChargeMetricsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetChargeMetricsQueryParams;

  @SpeakeasyMetadata()
  headers: GetChargeMetricsHeaders;
}


export class GetChargeMetricsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  chargeMetricsResponse?: shared.ChargeMetricsResponse;

  @SpeakeasyMetadata()
  commonReasonsErrorResponse?: shared.CommonReasonsErrorResponse;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;
}
